src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -76,29 +82,54 @@ object Gemini {
       //TODO: implement based on Apolo
       //findSimilarProjects(path, conn)
     } else {
-      findDuplicateFiles(path, conn)
+      findDuplicateItemForFile(path, conn)
       //TODO: implement based on Apolo
       //findSimilarFiles(path, conn)
     }
   }
 
   /**
-    * Finds duplicated files among hashed repositories that already exists in certain repository
+    * Finds duplicate files among hashed repositories
+    *
+    * @param conn     Database connection
+    * @param detailed If detailed is set to true, the output will contain which files are duplicated in each group
+    * @return
+    */
+  def report(conn: Session, detailed: Boolean): Iterable[Any] = {
+    val duplicates = findAllDuplicateItems(conn)
+    if (detailed) {
+      duplicates map { item =>
+        findDuplicateItemForBlobHash(item.sha, conn)
+      }
+    } else {
+      duplicates
+    }
+  }
+
+  /**
+    * Finds groups of duplicate files identified by the blob_hash
     *
-    * @param repository repository url, example: github.com/src-d/go-git.git"
+    * @param conn Database connection
     * @return
     */
-  def report(repository: String): Iterable[RepoFile] = {
-    throw new UnsupportedOperationException("Finding all duplicate files in many repositories is no implemented yet.")
+  def findAllDuplicateItems(conn: Session): Iterable[DuplicateBlobHash] = {
+    conn
+      .execute(new SimpleStatement(cql))
+      .asScala
+      .filter(_.getLong("count") > 1)
+      .map { r => DuplicateBlobHash(r.getString("blob_hash"), r.getLong("count")) }
```

Comment:
would you prefer keeping it like this, rather then

```scala
.map { r => 
  DuplicateBlobHash(r.getString("blob_hash"), r.getLong("count"))
}
```
?

It's not strong preference, this on just have advantage of being simpler to pattern-match by human eye.

Reply:
I'll try to apply your suggestion as a regular pattern in iterations while they're more than 80chars length. Thanks for noticing :dagger:

Reply:
I see! Two things
 - it may be a good idea to use this pattern to assist readably regardless of the length of the line
 - in this project, max line length is [120 chars](https://github.com/src-d/gemini/blob/master/scalastyle-config.xml#L17) which is friendlier to modern monitors

Reply:
I suggested the 80chars max for iterator callbacks to avoid the horizontal scroll in the GitHub interface; anyhow, I can take that rule for all callback iterators regardless its length.